base/linalg/uniformscaling.jl

```diff
@@ -31,6 +31,7 @@ eltype(::Type{UniformScaling{T}}) where {T} = T
 ndims(J::UniformScaling) = 2
 getindex(J::UniformScaling, i::Integer,j::Integer) = ifelse(i==j,J.λ,zero(J.λ))
 
+show(io::IO, J::UniformScaling{T}) where {T<:Complex} = print(io, "$(typeof(J))\n($(J.λ))*I")
```

Comment:
That was also my first version but I think negative reals become false positives here
```julia
julia> ismatch(r"\w+\s*\+|\-\s*\w+", "-1")
true
```